analyzer/src/main/scala/com/avsystem/commons/analyzer/ThrownExceptionNotInMonixScope.scala

```diff
@@ -0,0 +1,59 @@
+package com.avsystem.commons
+package analyzer
+
+import scala.tools.nsc.Global
+
+final class ThrownExceptionNotInMonixScope(g: Global) extends AnalyzerRule(g, "thrownExceptionNotInMonixScope") {
+
+  import global.*
+
+  lazy val monixTaskTpe: Type = classType("monix.eval.Task") match {
+    case NoType => NoType
+    case tpe => TypeRef(NoPrefix, tpe.typeSymbol, List(definitions.AnyTpe))
+  }
+  private def checkDiscardedNothing(tree: Tree, discarded: Boolean): Unit = tree match {
+    case tree if !discarded && tree.tpe != null && tree.tpe =:= definitions.NothingTpe =>
+      checkDiscardedNothing(tree, discarded = true)
+
+    case Block(stats: List[Tree], expr: Tree) =>
+      stats.foreach(checkDiscardedNothing(_, discarded = true))
+      checkDiscardedNothing(expr, discarded)
+
+    case Template(parents: List[Tree], self: ValDef, body: List[Tree]) =>
+      parents.foreach(checkDiscardedNothing(_, discarded = false))
+      checkDiscardedNothing(self, discarded = false)
+      body.foreach(checkDiscardedNothing(_, discarded = true))
+
+    case If(_: Tree, thenp: Tree, elsep: Tree) =>
+      checkDiscardedNothing(thenp, discarded)
+      checkDiscardedNothing(elsep, discarded)
+
+    case LabelDef(_: TermName, _: List[Ident], rhs: Tree) =>
+      checkDiscardedNothing(rhs, discarded = true)
+
+    case Try(block: Tree, catches: List[CaseDef], finalizer: Tree) =>
+      checkDiscardedNothing(block, discarded)
+      catches.foreach(checkDiscardedNothing(_, discarded))
+      checkDiscardedNothing(finalizer, discarded = true)
+
+    case CaseDef(_: Tree, _: Tree, body: Tree) =>
+      checkDiscardedNothing(body, discarded)
+
+    case Match(_: Tree, cases: List[CaseDef]) =>
+      cases.foreach(checkDiscardedNothing(_, discarded))
+
+    case Annotated(_: Tree, arg: Tree) =>
+      checkDiscardedNothing(arg, discarded)
+
+    case Typed(expr: Tree, _: Tree) =>
+      checkDiscardedNothing(expr, discarded)
+
+    case Apply(TypeApply(Select(prefix: Tree, TermName("map")), List(_)), List(Throw(_))) if prefix.tpe <:< monixTaskTpe =>
```

Comment:
Should we only handle mapping or consider other methods as well?

Reply:
I am gonna ask: should we also consider constructors?

Reply:
I don't understand the question: `Task(throw new RuntimeException("test"))` (not a constructor, but a factory) would work correctly in this case. 

In general, I'm not sure we should cover too many classes - let's start small - but I was thinking about the ones most closely related to the issue you were originally testing (e.g. `Task#flatMap`).

Reply:
I'd like to extend the scope, because in general we do not want to do `throw Ex` in place of possible `_ => throw Ex`, don't we? And yes, `Task.apply` is a function, but `new Sth(...)` is not and also we can cover this case, imo.